### PR TITLE
Add --fix-to-stdout

### LINKED ISSUE #52: Support --fix to stdout?
At the moment, `--fix` writes to the file. This makes it difficult to add as a pre-save hook in editors. There is no official support for writing the fixed output to stdout via the cli (see https://github.com/eslint/eslint/issues/6612) but it can be done w/ the api.

It'd be great to have `eslint_d` support that for the added speed boost, but it'd be a non-standard flag. Would you be up for having that in `eslint_d`?

### PR DESCRIPTION
**Pull request by aaronjensen**
fixes #52

### CODE REVIEW

File: lib/linter.js [hunk 1]
```diff
@@ -74,12 +74,19 @@ module.exports = function (cwd, args, text) {
 
     return JSON.stringify(fileConfig, null, '  ');
   }
+  if (currentOptions.fixToStdout && !stdin) {
```
Comment (aaronjensen) [inline lib/linter.js]: Should I check for `text` here? tbh, the check for text and the fallback to using files below is confusing. It seems `stdin` should *always* use stdin, even if the stdin is empty.

File: lib/linter.js [hunk 2]
```diff
@@ -74,12 +74,19 @@ module.exports = function (cwd, args, text) {
...
+      + '\n# exit 1';
+  }
   var report;
   if (stdin && text) {
     report = engine.executeOnText(text, currentOptions.stdinFilename);
   } else {
     report = engine.executeOnFiles(files);
   }
+  if (currentOptions.fixToStdout) {
+    return report.results[0].output || text;
```
Comment (aaronjensen) [inline lib/linter.js]: Related to above, there's an edge case here if a user passes `--stdin`, no text, and more than one file. I should probably at least check for `text` in the conditional, but that would introduce another edge case... if a user lint fixed a blank file, you'd want it to return a blank file (maybe with a trailing newline? 😄)

File: lib/options.js [hunk 3]
```diff
@@ -159,6 +159,12 @@ module.exports = optionator({
       description: 'Automatically fix problems'
     },
     {
+      option: 'fix-to-stdout',
+      type: 'Boolean',
+      description: 'Print the fix results to stdout instead of regular output, '
+        + 'must be used with --stdin'
```
Comment (aaronjensen) [inline lib/options.js]: Should we assume --stdin or just require it?

File: lib/linter.js (hunk 1 above)
Comment (mantoni) [inline lib/linter.js]: Hm, I see. Checking for `text` shouldn't be necessary. I agree that the flag should be enough.

File: lib/linter.js (hunk 2 above)
Comment (mantoni) [inline lib/linter.js]: Maybe the new option should only allow a single file if not used with `--stdin`? This way we don't need to require the `--stdin` option and one could use the feature to "inspect" what eslint does with `--fix` without actually modifying the file.

Comment (aaronjensen) [inline lib/linter.js]: is there an easy way to detect a single file? does eslint_d expand globs/directories or does it let eslint do that

Comment (mantoni) [inline lib/linter.js]: Globs are expanded by eslint. Let's take the easy path then and only support `--stdin` then. If text is empty, we just do nothing.

**Comment by aaronjensen** [comment]
@mantoni I think this is ready for review

**Comment by mantoni** [comment]
Looks good to me. Thanks a ton! 😎 Will merge and release later today or tomorrow …

**Comment by mantoni** [comment]
Released in `v4.2.0` (see [changelog for a description](https://github.com/mantoni/eslint_d.js/blob/master/CHANGES.md#420)).

**Comment by aaronjensen** [comment]
Nice. Unfortunately, I think that supporting non-stdin would be even trickier than knowing the file count. If there are no changes, I believe that eslint reports no source so we'd need to actually read the source from the file and send it back. Otherwise we could just count the # of results that came back and error if there are more than one.

**Comment by mantoni** [comment]
I think it was a good idea to make it work with a simple code path first. We can iterate on it once we have a good feeling about how this could work. I'm currently not a 100% sure about the approach we should take here. Maybe it's also not too important to have. Let's see.

File: README.md [hunk 4]
```diff
@@ -82,6 +84,42 @@ along with an access token in `~/.eslint_d`.
...
+hooks to automatically fix a file prior to saving. It must be used with
+`--stdin`.
+
+- __Emacs__: Add this to your `init.el`, be sure to add hooks for the actual
+  major modes you use:
+
+    ```elisp
+    (defun eslint-fix ()
+      (interactive)
+      (setq temp-point (point))
```
Comment (lunaryorn) [inline README.md]: @aaronjensen Why don't you `let`-bind this variable?  It's not a good idea to just `setq` variables this way.

Comment (lunaryorn) [inline README.md]: Actually you could just use `save-excursion` here, if I'm not mistaken.  That'll restore `point` automatically afterwards.  Did you try it?

Comment (aaronjensen) [inline README.md]: Unfortunately, `save-excursion` doesn't work w/ `shell-command-on-region`, that was the first I tried. Good idea on `let` though.

Comment (lunaryorn) [inline README.md]: Ah, too bad, didn't know that.

**Pull request merged**